src/main/java/io/confluent/copycat/hdfs/HdfsWriter.java

```diff
@@ -62,9 +80,14 @@ public HdfsWriter(HdfsSinkConnectorConfig connectorConfig) {
       if (!fs.exists(path)) {
         fs.mkdirs(path);
       }
-      offsets = readOffsets();
+      offsets = new HashMap<>();
+      readOffsets();
       writerProviderClass = (Class<RecordWriterProvider>)
-          Class.forName(connectorConfig.getString(HdfsSinkConnectorConfig.RECORD_WRITER_PROVIDER_CLASS_CONFIG));
+          Class.forName(connectorConfig.getString(
+              HdfsSinkConnectorConfig.RECORD_WRITER_PROVIDER_CLASS_CONFIG));
+
+      timedRotateThread = Executors.newSingleThreadScheduledExecutor();
```

Comment:
Do we need a separate thread here? Is the concern that if data isn't constantly flowing from the consumer that the connector might not be able to execute the timed rotation when it is supposed to. Is that something we should just have Copycat guarantee -- that regardless of available data you'll get a `put` call at least every X seconds or milliseconds -- even if it means calling `SinkTask.put` with an empty list of records sometimes? I'd like to avoid connectors allocating their own threads if at all possible.